### ISSUE TITLE
Roles done by parameterized roles not detectable via smartmatch

### ISSUE THREAD

**Issue by taboege** [rakudo/rakudo#2551]
## The Problem
Consider the code
``` perl6
role R { method f { "hi" } }
role S[$n] does R { }
say S[25] ~~ R;  # OUTPUT: False
say S[25].f;     # OUTPUT: hi
```
The first output tells me that `S[25]` doesn't `R`, but still `R` was apparently mixed it. Isn't this contradictory?

## Expected Behavior
It should tell me that `S[25]` does indeed do `R`. It works as expected with non-parameterized roles:
``` perl6
role R { method f { "hi" } }
role S does R { }
say S ~~ R;  # OUTPUT: True
say S.f;     # OUTPUT: hi
```

**Comment by vrurg** [comment]
Don't remember what PR closes this, but it is fixed since at least 2019.07.

**Comment by taboege** [comment]
Thanks! bisectable6 [tells me](https://gist.github.com/f9bc737dd76b7ac0048975c495996dde) that it might have been 41f6f9eaa.
